src/test/java/no/nav/tag/tiltaksgjennomforingprosess/JournalpostJobbTest.java

```diff
@@ -0,0 +1,102 @@
+
+    @Test
+    public void kallerIkkeJoarkMedEnAvtaleSomFeilet(){
+
+        Avtale okAvtale = TestData.opprettAvtale();
+        okAvtale.setId(okAvtale.getId());
+
+        Journalpost okJournalpost = new Journalpost();
+        okJournalpost.setEksternReferanseId(okAvtale.getId().toString());
+
+        Avtale feiletAvt = TestData.opprettAvtale();
+
+        Map<UUID, String> jorurnalpostIds = new HashMap<>(2);
+        final String JOURNALPOST_ID = "1234";
+        jorurnalpostIds.put(okAvtale.getId(), JOURNALPOST_ID);
+        jorurnalpostIds.put(feiletAvt.getId(), MAPPING_FEIL);
+
+        when(stsService.hentToken()).thenReturn("");
+        when(tiltaksgjennomfoeringApiService.finnAvtalerTilJournalfoering(anyString())).thenReturn(Arrays.asList(okAvtale, feiletAvt));
+
+        when(journalpostFactory.konverterTilJournalpost(okAvtale)).thenReturn(okJournalpost);
+        when(journalpostFactory.konverterTilJournalpost(feiletAvt)).thenThrow(HttpServerErrorException.class);
+
+        when(joarkService.sendJournalpost(anyString(), eq(okJournalpost))).thenReturn(JOURNALPOST_ID);
+
+        journalpostJobb.avtalerTilJournalfoering();
+        verify(joarkService, times(1)).sendJournalpost(anyString(), eq(okJournalpost));
+        verify(tiltaksgjennomfoeringApiService, atLeastOnce()).settAvtalerTilJournalfoert(anyString(), eq(jorurnalpostIds));
+    }
+
+    @Test
+    public void kallerMedAlleAvtaler(){
+
+        Avtale avtale1 = TestData.opprettAvtale();
+        Avtale avtale2 = TestData.opprettAvtale();
+
+        Journalpost journalpost1 = new Journalpost();
+        journalpost1.setEksternReferanseId(avtale1.getId().toString());
+        Journalpost journalpost2 = new Journalpost();
+        journalpost2.setEksternReferanseId(avtale2.getId().toString());
+
+
+        Map<UUID, String> jorurnalpostIds = new HashMap<>(2);
+        jorurnalpostIds.put(avtale1.getId(), avtale1.getId().toString());
+        jorurnalpostIds.put(avtale2.getId(), avtale2.getId().toString());
+
+        when(stsService.hentToken()).thenReturn("");
+        when(tiltaksgjennomfoeringApiService.finnAvtalerTilJournalfoering(anyString())).thenReturn(Arrays.asList(avtale1, avtale2));
+
+        when(journalpostFactory.konverterTilJournalpost(avtale1)).thenReturn(journalpost1);
+        when(journalpostFactory.konverterTilJournalpost(avtale2)).thenReturn(journalpost2);
+
+        when(joarkService.sendJournalpost(anyString(), eq(journalpost1))).thenReturn(avtale1.getId().toString());
+        when(joarkService.sendJournalpost(anyString(), eq(journalpost2))).thenReturn(avtale2.getId().toString());
+
+        journalpostJobb.avtalerTilJournalfoering();
+        verify(joarkService, times(1)).sendJournalpost(anyString(), eq(journalpost1));
+        verify(joarkService, times(1)).sendJournalpost(anyString(), eq(journalpost2));
+        verify(tiltaksgjennomfoeringApiService, atLeastOnce()).settAvtalerTilJournalfoert(anyString(), eq(jorurnalpostIds));
```

Comment:
Antar denne skal kjøre nøyaktig to ganger? Da kan det være `times(2)` i stedet for `atLeastOnce()`

Reply:
Nei, bare 1 gang.

Reply:
`times(1)` som er best da